lib/streamlit/runtime/connection_factory.py

```diff
@@ -0,0 +1,262 @@
+    useful, especially when working with a custom connection:
+
+    >>> import streamlit as st
+    >>> conn = st.connection("my_sql_connection", type="streamlit.connections.SQL")
+
+    Finally, you can even pass the connection class to use directly to this function.
+
+    >>> import streamlit as st
+    >>> from streamlit.connections import SQL
+    >>> conn = st.connection("my_sql_connection", type=SQL)
+    """
+    USE_ENV_PREFIX = "env:"
+
+    if name.startswith(USE_ENV_PREFIX):
+        # It'd be nice to use str.removeprefix() here, but we won't be able to do that
+        # until the minimium Python version we support is 3.9.
+        envvar_name = name[len(USE_ENV_PREFIX) :]
+        name = os.environ[envvar_name]
+
+    if type is None:
+        if name in FIRST_PARTY_CONNECTIONS:
+            # We allow users to simply write `st.experimental_connection("sql")`
+            # instead of `st.experimental_connection("sql", type="sql")`.
+            type = _get_first_party_connection(name)
+        else:
+            # The user didn't specify a type, so we try to pull it out from their
+            # secrets.toml file. NOTE: we're okay with any of the dict lookups below
+            # exploding with a KeyError since, if type isn't explicitly specified here,
+            # it must be the case that it's defined in secrets.toml and should raise an
+            # Exception otherwise.
+            secrets_singleton.load_if_toml_exists()
+            type = secrets_singleton["connections"][name]["type"]
+
+    # type is a nice kwarg name for the st.experimental_connection user but is annoying
+    # to work with since it conflicts with the builtin function name and thus gets
+    # syntax highlighted.
+    connection_class = type
+
+    if isinstance(connection_class, str):
+        # We assume that a connection_class specified via string is either the fully
+        # qualified name of a class (its module and exported classname) or the string
+        # literal shorthand for one of our first party connections. In the former case,
+        # connection_class will always contain a "." in its name.
+        if "." in connection_class:
+            parts = connection_class.split(".")
+            classname = parts.pop()
+            connection_module = importlib.import_module(".".join(parts))
+            connection_class = getattr(connection_module, classname)
+        else:
+            connection_class = _get_first_party_connection(connection_class)
+
+    # At this point, connection_class should be of type Type[ConnectionClass].
+    try:
+        return _create_connection(
+            name, connection_class, max_entries=max_entries, ttl=ttl, **kwargs
+        )
+    except ModuleNotFoundError as e:
+        err_string = str(e)
+        missing_module = re.search(MODULE_EXTRACTION_REGEX, err_string)
+
+        # TODO(vdonato): Finalize these error messages.
```

Comment:
nit: Here is still a todo :)

Reply:
Also removed in https://github.com/streamlit/streamlit/pull/6530